### ISSUE TITLE
Diamond does not set groups properly when changing uid and gid

### ISSUE THREAD

**Issue by nhooey** [python-diamond/Diamond#274]
According to [Python issue 10032](http://bugs.python.org/issue10032), setting the `uid` and `gid` is not enough to simulate logging in as that user. The `os.initgroups()` function needs to be used to initialize the groups of the user being changed to, like this:

```python
    if gid != -1 and uid != -1:
        os.initgroups(pwd.getpwuid(uid).pw_name, gid)
```

If this is not done, Diamond can't invoke actions on files when it changes to a user that only has group permissions for those files, since Python doesn't know that it's a member of those groups.

**Comment by nhooey** [comment]
This is fixed by #275.

**Comment by jaingaurav** [comment]
PR has been merged, thanks
